feature-cloud-backup-impl/src/main/java/io/novafoundation/nova/feature_cloud_backup_impl/data/cloudStorage/GoogleDriveBackupStorage.kt

```diff
@@ -0,0 +1,151 @@
+package io.novafoundation.nova.feature_cloud_backup_impl.data.cloudStorage
+
+import android.app.Activity.RESULT_OK
+import android.content.Intent
+import androidx.appcompat.app.AppCompatActivity
+import com.google.android.gms.auth.UserRecoverableAuthException
+import com.google.android.gms.auth.api.signin.GoogleSignIn
+import com.google.android.gms.auth.api.signin.GoogleSignInAccount
+import com.google.android.gms.auth.api.signin.GoogleSignInOptions
+import com.google.android.gms.common.api.ApiException
+import com.google.android.gms.common.api.Scope
+import com.google.android.gms.tasks.Task
+import com.google.api.client.googleapis.extensions.android.gms.auth.GoogleAccountCredential
+import com.google.api.client.googleapis.extensions.android.gms.auth.UserRecoverableAuthIOException
+import com.google.api.client.http.javanet.NetHttpTransport
+import com.google.api.client.json.gson.GsonFactory
+import com.google.api.services.drive.Drive
+import com.google.api.services.drive.DriveScopes
+import io.novafoundation.nova.common.resources.ContextManager
+import io.novafoundation.nova.common.resources.requireActivity
+import io.novafoundation.nova.common.utils.systemCall.SystemCall
+import io.novafoundation.nova.common.utils.systemCall.SystemCallExecutor
+import kotlinx.coroutines.Dispatchers
+import kotlinx.coroutines.withContext
+
+internal class GoogleDriveBackupStorage(
+    private val contextManager: ContextManager,
+    private val systemCallExecutor: SystemCallExecutor,
+    private val oauthClientId: String,
+    private val backupFileName: String = "novawallet_backup",
+) : CloudBackupStorage {
+
+    override suspend fun isUserAuthenticated(): Boolean = withContext(Dispatchers.IO) {
+        val account = GoogleSignIn.getLastSignedInAccount(contextManager.getApplicationContext())
+
+        account != null
+    }
+
+    override suspend fun authenticateUser(): Result<Unit> = withContext(Dispatchers.IO) {
+        val systemCall = GoogleSignInSystemCall(contextManager, oauthClientId, driveScope())
+        systemCallExecutor.executeSystemCall(systemCall)
+    }
+
+    override suspend fun checkBackupExists(): Result<Boolean> = withContext(Dispatchers.IO) {
+        runCatching { checkBackupExistsInner() }
+            .recoverCatching {
+                when (it) {
+                    is UserRecoverableAuthException -> it.askForConsent()
+                    is UserRecoverableAuthIOException -> it.cause?.askForConsent()
+                    else -> throw it
+                }
+
+                checkBackupExistsInner()
```

Comment:
For what reason do we check internal backups?

Reply:
`Inner` here just indicate private function that is wrapped by the public one. I will rename it to not cause confusion